app/Providers/ApiServiceProvider.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Providers;
+
+use App\Http\Resources\RegisterUserResource;
+use App\Services\ApiResponseService\ApiResponseBuilder;
+use Illuminate\Support\ServiceProvider;
+
+class ApiServiceProvider extends ServiceProvider
+{
+    /**
+     * Register services.
+     */
+    public function register(): void
+    {
+
+    }
+
+    /**
+     * Bootstrap services.
+     */
+    public function boot(): void
+    {
+        $this->app->bind('apiResponseFacade', function () {
+            return new ApiResponseBuilder();
+        });
+
+        $this->app->bind('RegisterUserResource', function ($resource, $params) {
+            return new RegisterUserResource($params['data']);
+        });
```

Comment:
I just started the topic of writing test.

and one point that was in the topic of writing test was that, if we use "new" for creating objects in the source code, we will face problems in the code where the "new"  is used, especially in mocking.

so, instead of using `new RegisterUserResource()` and passing parameter to it, i bind that to service container.

if I'm wrong, please tell me